### PR TITLE
ENH: Point cloud function and plotting function

### LINKED ISSUE #14: Point cloud
It would be great to have a function that takes a point cloud and a bunch of parameters and computes the regular and Steenrod barcode of it. A precursor for this is already present in the notebooks _cyclo-octane_ and _flat_Klein_bottle_.

### PR DESCRIPTION
**Pull request by ulupo**
Fixes #14 and (partly) #13 and refactors notebooks accordingly.

Important: introduces `GUDHI` and `plotly` requirements at installation time.

Removes `check_agreement_with_gudhi` function which had been once used for debugging.

Plotting is achieved in Plotly and it creates a 2D persistence diagram containing the regular bars and the Steenrod bars simultaneously.

### CODE REVIEW

**Comment by ulupo** [comment]
I'm rerunning all notebooks from scratch (the cyclo-octane one takes very long) so we can verify that the barcodes remain the same as in the paper.  Will post screenshots here.

**Comment by ulupo** [comment]
Klein bottle:

image
and
image

**Comment by ulupo** [comment]
Cyclo-octane:

image
and
image

@ammedmar please could you check that these and the ones for Klein look good, I've run out of time for now.

**Comment by ulupo** [comment]
@ammedmar I have pushed a plotting function, `plot_diagrams`.

You can test it on the Klein bottle notebook by adding the import line

```python
from steenroder.plotting import plot_diagrams
```

and then e.g. by plotting the result of the last computation (R=1.3, absolute cohomology) via

```python
plot_diagrams(barcode[:3], st_barcode[:3], k=1, kind="A")
```

File: steenroder/plotting.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+    return min_val_display, max_val_display, posinfinity_val, neginfinity_val
+
+
+def plot_diagrams(barcode, st_barcode, k=None, kind=None, plotly_params=None):
+    """Plot a regular persistence barcode and a Steenrod barcode as diagrams on
+    a common birth-death plane.
+
+    Parameters
+    ----------
+    bc : list of ndarray
```
Comment (gtauzin) [inline steenroder/plotting.py]: The parameter name is barcode.

**Review by gtauzin — CHANGES_REQUESTED** [review]
Lgtm! Just a small comment.

**Comment by ulupo** [comment]
Thanks @gtauzin!  I realized I forgot to document `rips_barcodes`, so I just pushed a simple docstring.

**Comment by ulupo** [comment]
Testing on @ammedmar's machine shows that the plotting function does not work well in all systems.  I think this is a MathJax-related issue because the function tries to produce LaTeX legend labels.  The simplest way to fix this would be to agree on default legend label names that do not require LaTeX, and still allow LaTeX rendering as an option in case people want to try that and it works.

@ammedmar what do you think?  What would be suitable names to replace $\mathcal{H}^i_A$, $\mathcal{H}^i_R$, $\mathrm{im}(Sq^k) \cap \mathcal{H}^i_A$, and $\mathrm{im}(Sq^k) \cap \mathcal{H}^i_R$, and that would not require LaTeX?

**Comment by ammedmar** [comment]
Yes, we can drop the LaTeX labels then. How about the following?
$\mathcal{H}^i_A$ --> H^i(abs)
$\mathcal{H}^i_R$ --> H^i(rel)
$\mathrm{im}(Sq^k) \cap \mathcal{H}^i_A$ --> im(Sq^k) in H^i(abs)
$\mathrm{im}(Sq^k) \cap \mathcal{H}^i_R$ --> im(Sq^k) in H^i(rel)

**Comment by ulupo** [comment]
@ammedmar I have now implemented your suggestion.  Could you please check that the diagrams in the Klein bottle notebook (in this branch) display as expected for you?  They display well at my end.

**Comment by ammedmar** [comment]
I also tested it on the rpn an cp2 notebooks and it looks good. You were right about tex being the issue.
Two very minor comments:
1. I realized that writing (abs) or (rel) in each line of the legend is maybe not ideal. Maybe one can just write absolute (relative) cohomology once as a title for the legend instead.
2. The legend doesn't seem to scale to fit below the diagonal (see screenshot below).

I don't think either of these needs to be address in this PR though.

**Comment by ammedmar** [comment]
plotting

**Comment by ammedmar** [comment]
Screenshot from 2022-03-25 09-30-48

**Review by ammedmar — APPROVED** [review]

**Comment by ulupo** [comment]
Thanks @ammedmar!

> Maybe one can just write absolute (relative) cohomology once as a title for the legend instead.

This is a good idea and would be one easy commit, I'll do it today.

> The legend doesn't seem to scale to fit below the diagonal (see screenshot below).

This is harder and I'd indeed leave it to another PR -- I just note that one can make the whole figure bigger as a temporary fix.

**Comment by ulupo** [comment]
@ammedmar I've made the change.  Let me know if happy to merge!

**Comment by ammedmar** [comment]
Yes, it looks good!

**Pull request merged**